src/Shared/ProviderConfiguration.cs

```diff
@@ -28,6 +29,7 @@ internal class ProviderConfiguration
         public int ConnectionTimeoutInMilliSec { get; set; }
         public int OperationTimeoutInMilliSec { get; set; }
         public string ConnectionString { get; set; }
+        public ISessionDataSerializer SessionDataSerializer { get; set; } = new DefaultSessionStateSerializer();
```

Comment:
This SessionDataSerializer property is not relevant to the ProviderConfiguration, right? If so, it shouldn't be added to this `ProviderConfiguration` shared by Session and Output cache providers. Instead, we should created dedicated `ProviderConfiguration`s for each (using a shared base class)

Reply:
Done